test/with_dummyserver/test_socketlevel.py

```diff
@@ -1289,12 +1289,12 @@ def socket_handler(listener):
 )
 class TestBrokenHeaders(SocketDummyServerTestCase):
 
-    def _test_broken_header_parsing(self, headers):
+    def _test_broken_header_parsing(self, headers, unparsed_data_check=None):
```

Comment:
Why were these test cases changed?

Reply:
Do you mean the addition of unparsed_data_check specifically?

I looked at the various Stackoverflow questions where this had come up, and realised we already had a test case where unparsed_data was non-empty, so I thought it would be good to check for that.

Reply:
I meant more the input data, it looks like you've changed the number of CRLF within the test case, what was the motivation for those changes?

Reply:
As for the changes in the values of `headers` passed into `_test_broken_header_parsing()` and the subsequent generation of the value to pass into `self.start_response_handler()`, here's what two of the test cases were actually checking:

```py
b'HTTP/1.1 200 OK\r\nContent-Length: 0\r\nContent-type: text/plain\r\n: Value\r\n\r\nAnother: Header\r\n\r\n'
b'HTTP/1.1 200 OK\r\nContent-Length: 0\r\nContent-type: text/plain\r\n:\r\n\r\nAnother: Header\r\n\r\n'
```

The `'\r\n\r\n'` before `Another: Header` meant that `Another: Header` wasn't a part of the headers anymore, and wasn't being checked. In all the present test cases that didn't affect the result of the test, but that was more by coincidence.

Reply:
What's next for this PR? Are you happy with my explanations, or would you prefer me to rework the tests?